units/src/amount.rs

```diff
@@ -2075,6 +2105,25 @@ mod tests {
         }
     }
 
+    #[test]
+    fn from_str_zero_without_denomination() {
+        let _a = Amount::from_str("0").unwrap();
+        let _a = Amount::from_str("0.0").unwrap();
+        let _a = Amount::from_str("00.0").unwrap();
+
+        assert!(Amount::from_str("-0").is_err());
+        assert!(Amount::from_str("-0.0").is_err());
+        assert!(Amount::from_str("-00.0").is_err());
+
+        let _a = SignedAmount::from_str("-0").unwrap();
+        let _a = SignedAmount::from_str("-0.0").unwrap();
+        let _a = SignedAmount::from_str("-00.0").unwrap();
+
+        let _a = SignedAmount::from_str("0").unwrap();
+        let _a = SignedAmount::from_str("0.0").unwrap();
+        let _a = SignedAmount::from_str("00.0").unwrap();
+    }
```

Comment:
A test confirming that `0.00` parses would be nice too.

Reply:
can just do in another comment since this one is already acked.  This is also something that could just be part of a proptest given arbitrary dot placement which is probably more effective.

Reply:
Good idea turning it to proptest, but I'd like to see also various lengths not just dot placement being changed.